build.zig

```diff
@@ -20,7 +20,7 @@ pub fn build(b: *Build) void {
 
     const lang = b.option(Language, "lang", "Lua language version to build") orelse .lua54;
     const shared = b.option(bool, "shared", "Build shared library instead of static") orelse false;
-
+    const luau_vector_size = b.option(usize, "luau_vector_size", "Luau built-in vector size: 3-vector or 4-vector") orelse 3;
```

Comment:
I'm not sure, but do you think it would be reasonable to use an enum for this? `enum { @"3", @"4" }`? This would enforce that only those integers are used.

An alternative is to name this `"luau_use_4_vector"` or something similar, and make it a bool.

Or perhaps making a compileError when an invalid number is passed.

Looking through the rest of the code, I see that `luau_vector_size` is sometimes used as a value (like the size of the array), so there is also a benefit to keeping this a usize.

Reply:
I actually tried to add a `@compileError` in build.zig for this but somehow I couldn't get it to work.  It was getting late and I was tired.. I'll try again tonight.  :) 

I originally had `luau_use_4_vector` as a bool, like you noticed below.

IMO at least this public export 

```
pub const luau_vector_size = config.luau_vector_size;
```

should be an integer, so that you can directly use that for example to size slices correctly.  An enum would be a bit cumbersome.

Reply:
Yeah, adding

```
    if (luau_vector_size != 3 or luau_vector_size != 4) {
        @compileError("luau_vector_size must be 3 or 4");
    }
```
in the build.zig build function always triggers a `@compileError`.  I guess only `@panic` can be used in build.zig.

I'll try a bool arg instead.